### PR TITLE
nvme: add support for lba status log page

### PR DESCRIPTION
**Pull request by ganaidu707**
This log page is used to provide information about
subsequent actions the host may take to discover which
logical blocks, in namespaces that are attached to
the controller, may no longer be recoverable when read.
For more details see NVM Express 1.4 Spec. Section
5.14.1.14("LBA Status Information (Log Identifier 0Eh)")

Signed-off-by: Gollu Appalanaidu <anaidu.gollu@samsung.com>
Co-Authored-By: Karthik Balan karthik.b82@samsung.com

### CODE REVIEW

File: linux/nvme.h [hunk 1]
```diff
@@ -713,6 +713,29 @@ struct nvme_effects_log {
...
+	struct nvme_lba_range_desc desc[];
+};
+
+struct nvme_lba_status_log_page {
+	__le32	lslplen;
+	__le32	nlslne;
+	__le32	estulb;
+	__u8	rsvd12[2];
+	__le16	lsgc;
+	struct nvme_lba_status_ns_element element;
```
Comment (keithbusch) [inline linux/nvme.h:736]: This is not a single element. It is a variable length array of size nlslne.

Comment (ganaidu707) [reply linux/nvme.h]: This has been addressed and rebased it!

File: nvme-print.c [hunk 2]
```diff
@@ -1614,6 +1614,103 @@ void nvme_show_endurance_group_event_agg_log(
...
+		le32_to_cpu(lba_status->lslplen));
+        num_elements = le32_to_cpu(lba_status->nlslne);
+	json_object_add_value_uint(root, "nlslne",
+		num_elements);
+	json_object_add_value_uint(root, "estulb",
+		le32_to_cpu(lba_status->estulb));
+	json_object_add_value_uint(root, "lsgc",
+		le16_to_cpu(lba_status->lsgc));
+
+        elements_list = json_create_array();
```
Comment (keithbusch) [inline nvme-print.c:1637]: Tabs, not spaces please.

File: nvme-print.c [hunk 3]
```diff
@@ -1614,6 +1614,103 @@ void nvme_show_endurance_group_event_agg_log(
...
+        for (int ele = 0; ele < num_elements; ele++) {
+                element = json_create_object();
+                json_object_add_value_uint(element, "neid",
+                        le32_to_cpu(lba_status->elements[ele].neid));
```
Comment (keithbusch) [inline nvme-print.c:1641]: I'm terribly confused how this even compiles. We have a variable length array who's elements are variable in size. How could the compiler possibly know the address of the element index?

Comment (birkelund) [reply nvme-print.c:1641]: You need -Wpedantic to complain about this. I'm not sure if it is undefined behavior.

Comment (keithbusch) [reply nvme-print.c:1641]: The behavior assumes the elements are fixed length `sizeof(lba_status->elements[0])`, so accessing anything except for the first index is going to yield the wrong address.

Comment (ganaidu707) [reply nvme-print.c:1641]: I have addressed all the review comments and raised it as separate PR https://github.com/linux-nvme/nvme-cli/pull/933
will close this PR.
Thanks!

**Pull request closed without merging**